test/dict.jl

```diff
@@ -374,6 +376,32 @@ let
     Base.show(Base.IOContext(IOBuffer(), :limit => true), a)
 end
 
+let
+    a = ObjectIdDict()
+    a[1] = a
+    a[a] = 2
```

Comment:
I've addressed all the comments but this one - what's a reasonable looking test for this?